src/backend/utils.ts

```diff
@@ -1288,6 +1290,69 @@ async function getPathDiskSize(path: string): Promise<number> {
   return statData.size
 }
 
+export async function checkRosettaInstall() {
+  if (!isMac) {
+    return
+  }
+
+  // check if on arm64 macOS
+  const { stdout: archCheck } = await execAsync('arch')
+  const isArm64 = archCheck.trim() === 'arm64'
```

Comment:
why not use the `isIntelMac` constant here? then we don't need to execute a process

if it's not intel mac then it's arm